src/sentry/static/sentry/app/views/projectGeneralSettings.jsx

```diff
@@ -74,9 +88,33 @@ export default class ProjectGeneralSettings extends AsyncView {
     return val + ' hour' + (val != 1 ? 's' : '');
   }
 
-  renderRemoveProject() {
-    let {orgId, projectId} = this.props.params;
+  handleTransferFieldChange = (id, value) => {
+    this._form[id] = value;
+  };
+
+  handleRemoveProject = () => {
+    let {orgId} = this.props.params;
+    let project = this.state.data;
+    if (!project) return;
 
+    removeProject(this.api, orgId, project).then(() => {
+      // Need to hard reload because lots of components do not listen to Projects Store
+      window.location.assign('/');
+    });
+  };
+
+  handleTransferProject = () => {
+    let {orgId} = this.props.params;
+    let project = this.state.data;
+    if (!project) return;
+    if (!this._form.email) return;
```

Comment:
Could we disable the transfer button instead so they don't end up here?

Reply:
Hmm actually this is not as easy because of the Confirm component